Clarify how you would end up with None in programming.rst (#130498)
python/cpython: Doc/faq/programming.rst

and :func:`sorted(y) <sorted>`) create a new object.  In general in Python (and in all cases
in the standard library) a method that mutates an object will return ``None``
to help avoid getting the two types of operations confused.  So if you

> mistakenly write ``z=y.sort()`` thinking it will give you a sorted copy of ``y``,

Reviewer: ```suggestion
mistakenly write ``z = y.sort()`` thinking it will give you a sorted copy of ``y``,
```

> mistakenly write ``z = y.sort()`` thinking it will give you a sorted copy of ``y``,
> you'll instead end up with ``None``, which will likely cause your program to
> generate an easily diagnosed error.

Reviewer 2: ```suggestion
mistakenly write ``z = y.sort()`` thinking ``z`` is a sorted copy of ``y``,
you'll instead end up with ``z`` being ``None``, which will likely cause
your program to generate an easily diagnosed error.
```
